zap/src/main/java/org/zaproxy/zap/eventBus/SimpleEventBus.java

```diff
@@ -203,78 +204,54 @@ public void publishSyncEvent(EventPublisher publisher, Event event) {
-                    "Publisher not registered: " + publisher.getPublisherName());
-        }
-        log.debug(
-                "publishSyncEvent "
-                        + event.getEventType()
-                        + " from "
-                        + publisher.getPublisherName());
-        boolean foundType = false;
-        for (String type : regPublisher.getEventTypes()) {
-            if (event.getEventType().equals(type)) {
-                foundType = true;
-                break;
-            }
+            throw new InvalidParameterException("Publisher not registered: " + publisherName);
         }
-        if (!foundType) {
+
+        log.debug("publishSyncEvent " + eventType + " from " + publisherName);
+        if (!regPublisher.getEventTypes().contains(eventType)) {
             throw new InvalidParameterException(
-                    "Event type: "
-                            + event.getEventType()
-                            + " not registered for publisher: "
-                            + publisher.getPublisherName());
+                    "Event type: " + eventType + " not registered for publisher: " + publisherName);
         }
 
-        for (RegisteredConsumer regCon : regPublisher.getConsumers()) {
-            String[] eventTypes = regCon.getEventTypes();
-            boolean isListeningforEvent = false;
-            if (eventTypes == null) {
-                // They are listening for all events from this publisher
-                isListeningforEvent = true;
-            } else {
-                for (String type : eventTypes) {
-                    if (event.getEventType().equals(type)) {
-                        isListeningforEvent = true;
-                        break;
-                    }
-                }
-            }
-            if (isListeningforEvent) {
-                try {
-                    regCon.getConsumer().eventReceived(event);
-                } catch (Exception e) {
-                    log.error(e.getMessage(), e);
-                }
-            }
-        }
+        regPublisher
+                .getConsumers(eventType)
+                .forEach(
+                        regCon -> {
+                            try {
+                                regCon.getConsumer().eventReceived(event);
+                            } catch (Exception e) {
+                                log.error(e.getMessage(), e);
+                            }
+                        });
     }
 
     private static class RegisteredConsumer {
         private EventConsumer consumer;
-        private String[] eventTypes;
+        private Set<String> eventTypes;
         private String publisherName;
 
-        public RegisteredConsumer(EventConsumer consumer, String[] eventTypes) {
+        RegisteredConsumer(EventConsumer consumer, Set<String> eventTypes) {
             this.consumer = consumer;
-            this.eventTypes = eventTypes;
+            this.eventTypes = unmodifiableSet(eventTypes);
```

Comment:
We don't need to, no, but I thought it would be nice to make our intentions clear. It still irks me that the immutability ot collections isn't part of the type system :(

Reply:
I think it's implied/clear that's not intended to be modified (why would we change the event types that the consumer wants?).

Reply:
Actually there are no methods on `RegisteredConsumer` that let you mutate it anyway so yeah, it can go